### ISSUE TITLE
Visual Studio Test using Test Agent task will fail build if it contains skipped tests

### ISSUE THREAD

**Issue by Framnk** [Microsoft/vsts-tasks#2055]
Our unit test assemblies contain several tests that are flagged as skipped (for various reasons).  However using the Visual Studio Test using Test Agent task in a build configuration will flag the build as as a failure if any unit tests are skipped during the test run and there is no option on this task to ignore skipped tests.  Note that under Test Runs it will properly show a warning showing that some tests were skipped during the test run.

2016-07-11T14:29:05.8124192Z DistributedTests: Test run '2399' is in 'Completed' state.
 
2016-07-11T14:29:05.8129194Z DistributedTests: Total Tests : 1443, Passed Tests : 1311
 
2016-07-11T14:29:15.8249226Z ##[error]Some tests in the test run did not pass, failing the task.

![image](https://cloud.githubusercontent.com/assets/1591272/16735527/58a66a24-4757-11e6-8525-44ede12a50f8.png)

**Comment by Framnk** [comment]
Note the 'failed' status (versus warning) in the screenshot below:

![image](https://cloud.githubusercontent.com/assets/1591272/16735599/ae23864e-4757-11e6-8b76-5071eb47a309.png)

**Comment by tanvi-soni** [comment]
Can you connect to the sql database for the tfs and query the tbl_testrun for above run id. Please provide me the values of following columns in tbl_testrun: TotalTests, PassedTests, NotApplicableTests.

**Comment by Framnk** [comment]
I am not the TFS administrator so I do not have direct access to the sql database.  I've requested this information but it might take some time.

**Comment by vimegh** [comment]
@Framnk Thanks for asking your TFS Administrator for the data, we can wait for that. In the meanwhile can you please share with us the trx/result files that is generated for this test run, it seems to us that one failed test is not getting listed to you on UI. 

Also In Build->Tests page, if you set outcome as failed or Others, do you see any entries, can you please click a screenshot and send to us.

**Comment by Framnk** [comment]
Unfortunately I'm blocked in gathering the results of the test run due to the other issue I've logged regarding the Test Agent Deployment not working with 'Interactive Process' checked.  Prior to Update 3 I was able to work around the problem by running a separate build config with the Test Agent Deployment task, cancel the build then run a separate build config with the automation tests.  However with update 3 it now _requires_ that the Test Agent Deployment task succeed before attempting to run the unit test task.
